docs/src/main/asciidoc/secret-management.adoc

```diff
@@ -14,165 +19,188 @@ spring-cloud-azure-starter-keyvault-secrets adds Azure Key Vault as one of the S
+| *spring.cloud.azure.keyvault.secret*.property-sources[].endpoint        |                                  | Azure Key Vault endpoint.
+| *spring.cloud.azure.keyvault.secret*.property-sources[].case-sensitive  | false                            | Whether the secret keys are case-sensitive.
+| *spring.cloud.azure.keyvault.secret*.property-sources[].secret-keys     |                                  | The secret keys supported for this property source. All keys be retrieved if this property is not configured.
+| *spring.cloud.azure.keyvault.secret*.property-sources[].refresh-interval| 30m                              | Time interval to refresh all Key Vault secrets.
+| *spring.cloud.azure.keyvault.secret*.property-sources[].service-version |                                  | Secret service version used when making API requests.
+| *spring.cloud.azure.keyvault.secret*.property-sources[].client          |                                  | Client related properties.
+| *spring.cloud.azure.keyvault.secret*.property-sources[].credential      |                                  | Credential related properties.
+| *spring.cloud.azure.keyvault.secret*.property-sources[].profile         |                                  | Profile related properties.
+| *spring.cloud.azure.keyvault.secret*.property-sources[].proxy           |                                  | Proxy related properties.
+| *spring.cloud.azure.keyvault.secret*.property-sources[].retry           |                                  | Retry related properties.
+|===
+
+- Please refer to link:index.html#authorize-access-with-azure-active-directory[Authorize access with Azure AD] to make sure the link:https://docs.microsoft.com/azure/active-directory/develop/app-objects-and-service-principals#service-principal-object[security principal] has been granted the sufficient permission to access the Azure Key Vault secrets.
+- If common properties like `client`, `credential`, `profile`, `proxy`, `retry` are not configured in `spring.cloud.azure.keyvault.secret.property-sources[].xxx`, `spring.cloud.azure.xxx` will be used. Please refer to link:index.html#configuration[Configuration] to get more information about these common properties.
+- Please refer to link:appendix.html#_configuration_properties[Configuration Properties] to get more information about nested properties.
+
+==== Samples
+
+Sample project: link:https://github.com/Azure-Samples/azure-spring-boot-samples/tree/spring-cloud-azure_{project-version}/keyvault/spring-cloud-azure-starter-keyvault-secrets/property-source[property-source].
+
+
+=== Auto Configure beans for Azure Key Vault Secret Clients
+
+==== Dependency Setup
+
+[source,xml]
+----
+<dependency>
+    <groupId>com.azure.spring</groupId>
+    <artifactId>spring-cloud-azure-starter-keyvault-secrets</artifactId>
+</dependency>
+----
+
+==== Basic Usage
+
+===== Configuration Properties
```

Comment:
This part is only about secret management, and there's already a part for autoconfiguring the SDK client https://microsoft.github.io/spring-cloud-azure/current/reference/html/index.html#auto-configure-azure-sdk-clients

Reply:
And it's better to have a central place to introduce all features provided by `spring-cloud-azure-starter-keyvault-secrets`.

It's not convenient to customer to understand the feature of `SecretClient` by browsing multiple sections in this doc:
1. https://microsoft.github.io/spring-cloud-azure/current/reference/html/index.html#auto-configure-azure-sdk-clients
2. https://microsoft.github.io/spring-cloud-azure/current/reference/html/index.html#configuration

Especially that these sections didn't link to each other. It's hard for our customer to understand that `spring.cloud.azure.keyvault.secret.client.logging.allowed-header-names=User-Agent` can be used to configure the autoconfigured `SecretClient` bean. 

Here is my current suggestion:
1. Enrich current content in [auto-configure-azure-sdk-clients](https://microsoft.github.io/spring-cloud-azure/current/reference/html/index.html#auto-configure-azure-sdk-clients) and [configuration](https://microsoft.github.io/spring-cloud-azure/current/reference/html/index.html#configuration)
2. For sections like `secret management`, provide a simple introduce to the `auto configured client` feature, and have link to related section.

Reply:
And keep both the 2 sections can make it clear why there are 2 properties about `endpoint`: `spring.cloud.azure.keyvault.secret.endpoint` and `spring.cloud.azure.keyvault.secret.property-sources[].endpoint`. Delete it may make customer confused.